lib/shrine/storage/memory.rb

```diff
@@ -17,7 +17,10 @@ def upload(io, id, *)
       end
 
       def open(id, *)
-        StringIO.new(store.fetch(id))
+        str = store.fetch(id)
+        # work around apparent bug in ruby 2.7.0
+        # https://bugs.ruby-lang.org/issues/16497
+        StringIO.new(str).set_encoding(str.encoding, str.encoding)
```

Comment:
I have to say I don't fully understand what internal vs external encoding do exactly, so I would prefer if we can only make minimal changes necessary.

On my computer, I see the following:

```rb
require "stringio"

io = StringIO.new("foo".b)
p [io.internal_encoding, io.external_encoding]

# ruby 2.6.5:
# [nil, #<Encoding:ASCII-8BIT>]

# ruby 2.7.0
# [nil, #<Encoding:ASCII-8BIT>]
```

When I add `Encoding.default_internal = Encoding::UTF_8`, I see the following:

```rb
require "stringio"

Encoding.default_internal = Encoding::UTF_8

io = StringIO.new("foo".b)
p [io.internal_encoding, io.external_encoding]

# ruby 2.6.5:
# [nil, #<Encoding:ASCII-8BIT>]

# ruby 2.7.0
# [nil, #<Encoding:UTF-8>]
```

So, it seems that changing the `Encoding.default_internal` changes the StringIO's *external* encoding 🤷‍♂ (which I see mentioned in the bug report). However, in that case I prefer only changing the StringIO's external encoding, and keeping the internal one as is, because we don't actually need to change it. We can do this by passing a single argument to `#set_encoding`.

Reply:
OK, it still passes the test i added with only the single arg, so I guess that does work! 

(Always interesting how many hours of research it takes to get to one line of code sometimes! This was a very tricky one to understand, I still don't totally understand it).